test/test.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const rehype = require('rehype');
+const dedent = require('dedent');
+const rehypeHighlightCodeBlock = require('..');
+
+describe('rehypeHighlightCodeBlock', () => {
+  let mockHighLightSpy;
+  let processHtml;
+
+  beforeEach(() => {
+    mockHighLightSpy = jest.fn(() => '<span>mockSyntaxHighlighting</span>');
+    processHtml = html => {
+      return rehype()
+        .data('settings', { fragment: true })
+        .use(rehypeHighlightCodeBlock, {
+          highlight: mockHighLightSpy
+        })
+        .processSync(html)
+        .toString();
+    };
+  });
+
+  test('mocks syntax highlighting with language-* class', () => {
+    const result = processHtml(dedent`
+      <div>
+        <p>foo</p>
+        <pre><code class="language-css">p { color: red }</code></pre>
+        <p>bar</p>
+      </div>
+    `);
+    expect(mockHighLightSpy.mock.calls[0].length).toBe(2);
+    expect(mockHighLightSpy.mock.calls[0][1]).toBe('css');
```

Comment:
I think we should be able to make an assertion about both arguments. This line might be able to replace 32-33:

```js
expect(mockHighlightSpy).toHaveBeenCalledWith(
  `p { color: red }`,
  'css'
);
```

We need to assert that the correct first argument, the content of the code block, is passed in.